components/constellation/constellation.rs

```diff
@@ -5417,6 +5413,44 @@ where
             }
         }
     }
+    /// Handle GamepadEvents from the embedder and forward them to the script thread
+    #[cfg_attr(
+        feature = "tracing",
+        tracing::instrument(skip_all, fields(servo_profiling = true), level = "trace")
+    )]
+    fn handle_theme_change(&mut self, theme: Theme) {
+        // Send to the focused browsing contexts' current pipeline.
+        let focused_browsing_context_id = self
```

Comment:
I don't know what is the correct way to get the browser context id or pipeline id or how they work in general when passing events. Any guidance here would be helpful

Reply:
@jdm Would you know how best to thread this though here? I know nothing of browsing contexts or the script/constellation code. The high-level aim here is to pass winit (or other embedder) "theme changed" events through to the layout thread so that it can enable/disable "dark mode" (`prefers-color-scheme`) in Stylo.

Reply:
You can follow the model of [handle_broadcast_storage_event](https://github.com/servo/servo/blob/main/components/constellation/constellation.rs#L2596), which also needs to communicate with multiple script threads. By sending a new [ConstellationControlMsg](https://github.com/servo/servo/blob/3f69ef2303dd227c49917c1691e841dca41a4ad2/components/shared/script/lib.rs#L275) value, you can have the script thread [process the new message](https://github.com/servo/servo/blob/3f69ef2303dd227c49917c1691e841dca41a4ad2/components/script/script_thread.rs#L2267) and [iterate over the known documents](https://github.com/servo/servo/blob/3f69ef2303dd227c49917c1691e841dca41a4ad2/components/script/script_thread.rs#L1318-L1320) to obtain a [Layout](https://github.com/servo/servo/blob/3f69ef2303dd227c49917c1691e841dca41a4ad2/components/script/dom/window.rs#L368) object from the document's associated window.

Reply:
(I'm assuming we actually want to broadcast this to all known documents, rather than just the active one in the focused webview)

Reply:
> (I'm assuming we actually want to broadcast this to all known documents, rather than just the active one in the focused webview)

I think so, at least for now. Separate themes per-window is technically possible I think, but a very rare edge case in practice.

@arthmis Can you update this PR as per jdm's suggestions above? (also, there are some copy-pasted comments that you should update to reflect your new code, and some debug statements which should be removed before we land this).